### DISCUSSION TITLE
How to create a new digital product, steps?

### DISCUSSION THREAD

**Discussion by g-londhe** [saleor/saleor#9482] [Q&A] [answered]
Hi,
I'm trying to create a digital product(Method 1) from API as suggested here -> 
(https://github.com/saleor/saleor/issues/4682#issuecomment-527320768)

1) What is the digital variant in step 4 and where do I get that from?

Also, I'm trying to create a digital product(Method 2) with the following steps from API ->
1) Create a product,
2) Update product channel listing,
3) Create a product variant,
4) Update Product variant channel listing
5) Create digital content
6) Shop Settings - Set auto fulfillment for digital products to true

When I try to checkout for the digitally created product, my order is not being automatically fulfilled, I would like to know more about it, in case I'm missing something.

Sharing the queries below in the input.txt file for more info.
[input.txt](https://github.com/saleor/saleor/files/8418636/input.txt)

**Comment by karolkielecki** [comment]
Hi @g-londhe, could you please tell us your Saleor version? The comment you are referring to might not be relevant anymore--since then we released Saleor 2.11, then 3.0 and recently 3.1 with many breaking changes, improvements, etc. You can easily create a digital product through the dashboard. An example of digital product variant is here: https://demo.saleor.io/dashboard/products/UHJvZHVjdDoxMjI%3D/variant/UHJvZHVjdFZhcmlhbnQ6MzE5

**Reply by g-londhe** [reply]
Hi @karolkielecki,

1. I'm trying to create a product from an external API call,
2. Currently I'm using **dashboard v3.1.0-a.18 core v3.1.0-a.25** that comes up with Saleor docker.
3. Could you please share the steps I need to follow in order to create the digital product?
That would be helpful. pls, refer to the steps(Input.txt) that I have followed while creating a product.

**Reply by bandieramonte** [reply]
Hi @karolkielecki, thank you for your input. I just checked the example digital variant from the link you sent, and need to inquire as follows:
- the option to make it non shippable is available only to product types, and not to variants. This means that it is not possible to create a product having both the physical and digital variants (e.g. a book product that can be bought as hardcover or Ebook). One would need to add 2 separate products associated to the same book. Am I missing something?  

- I can't find in the Dashboard where one can upload the file that users can download upon order completion for digital products. Is this feature implemented or any ideas when this would be implemented? Otherwise, would the best way to implement this be to create a new app and integrate it within the Dashboard?

**Comment by robert74j** [comment]
Hi @karolkielecki,

How to upload a digital content e.g. a pdf file for a digital product already defined ? Is there any way to do this using dashboard ?

**Reply by fureunoir** [reply]
Hi! As I can see, there's no options to upload digital content via dashboard, but my team forked the repo and now we're working on it. As soon as we reach anything I'll let people know here.

**Reply by timuric** [reply, marked as answer]
You can create file attribute for products to upload content

**Reply by bandieramonte** [reply]
@timuric when creating a file attribute for products to upload content and adding such attribute to a product type, the dashboard does not let me use such attribute as a variant selection attribute.
Is it possible somehow to add different files to different variants of a product, so that the user can later download the corresponding file only after paying for the variant?

**Reply by timuric** [reply]
Yes, it possible to add attributes to variants, like so, without selecting "selection variant" checkbox
<img width="1598" alt="image" src="https://github.com/saleor/saleor/assets/4006792/44309812-d2d0-4447-bc43-693946cc3152">

**Reply by bandieramonte** [reply]
@timuric if I do as you say, then I'm not able to add a file attribute per variant. This file attribute simply does not show when creating a new variant for a product. I just tried this now in the dashboard. If you can show me with a pic how you add a file attribute per variant, or if you have documentation for this, I would be happy to see this.

**Reply by timuric** [reply]
https://github.com/saleor/saleor/assets/4006792/4c1f9903-b5c2-4437-b2c5-664236e5c3ff

**Reply by bandieramonte** [reply]
@timuric thank you. I had done before everything you showed in the video. 
Now, can you try adding a new product of that type (Default Type) and create 2 variants, assigning different files to each variant?
This is the part I cannot do. If you are able to do this, I would appreciate to see a video of this.
